src/components/shared/DetailsWidget/TeamDetail/TeamDetail.tsx

```diff
@@ -1,22 +1,34 @@
 import React from 'react';
 
 import ColorTag from '~shared/ColorTag';
-import { DomainColor, Domain } from '~types';
+import { DomainMetadata } from '~types';
 
 import styles from './TeamDetail.css';
 
 const displayName = 'DetailsWidget.TeamDetail';
 
 interface TeamDetailProps {
-  domain: Domain;
+  transactionHash?: string;
+  domainMetadata: DomainMetadata;
 }
 
-const TeamDetail = ({ domain }: TeamDetailProps) => (
-  <div>
-    <ColorTag color={domain?.metadata?.color ?? DomainColor.LightPink} />
-    <span className={styles.text}>{` ${domain?.metadata?.name}`}</span>
-  </div>
-);
+const TeamDetail = ({ transactionHash, domainMetadata }: TeamDetailProps) => {
+  const changelogItem = transactionHash
+    ? domainMetadata.changelog?.find(
+        (item) => item.transactionHash === transactionHash,
+      )
+    : undefined;
+
+  const domainColor = changelogItem?.newColor ?? domainMetadata.color;
+  const domainName = changelogItem?.newName ?? domainMetadata.name;
```

Comment:
Could this logic to get domainColor and domainName be moved into a utility function?